### PR TITLE
Refactor TweetSearcherServer

### PR DESCRIPTION
**Pull request by xeniaqian94**
Refactor with Issue#13 Jetty servlet

### CODE REVIEW

File: src/main/java/io/anserini/nrts/TweetServlet.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.TopScoreDocCollector;
+//import org.eclipse.jetty.http.HttpStatus;
+
+import io.anserini.nrts.TweetStreamIndexer.StatusField;
+
+public class TweetServlet extends HttpServlet {
+	private static int topN = 20;
```
Comment (lintool) [inline src/main/java/io/anserini/nrts/TweetServlet.java]: Please clean up code formatting - indentation = 2 spaces
unless objections, see https://github.com/lintool/Anserini/issues/16

File: src/main/java/io/anserini/nrts/TweetSearcher.java [hunk 2]
```diff
@@ -68,10 +73,22 @@ public static void main(String[] args) throws IOException, InterruptedException,
...
     LOG.info("Starting HTTP server on port " + port);
-    TweetSearcherServer webServer = new TweetSearcherServer(port);
-    webServer.start();
+    
+    
+    Server server = new Server(port);
+    ServletHandler handler = new ServletHandler();
+    server.setHandler(handler);
+    handler.addServletWithMapping(TweetServlet.class, "/*");
+    try {
```
Comment (lintool) [inline src/main/java/io/anserini/nrts/TweetSearcher.java]: Please clean up code formatting - indentation = 2 spaces
unless objections, see https://github.com/lintool/Anserini/issues/16

File: pom.xml [hunk 3]
```diff
@@ -156,6 +162,18 @@
       <artifactId>guava</artifactId>
       <version>19.0-rc2</version>
     </dependency>
+    
+    <dependency>
+  		<groupId>org.eclipse.jetty</groupId>
```
Comment (lintool) [inline pom.xml]: fix XML indentation please,
add variable as in ${LUCENE_VERSION} for Jetty so you can reference to versions consistently in one place.

File: src/main/java/io/anserini/nrts/TweetServlet.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+	private static final Logger LOG = LogManager.getLogger(TweetServlet.class);
+	private IndexReader reader;
+	
+	@Override
+	protected void doGet(HttpServletRequest request, HttpServletResponse response)
+			throws ServletException, IOException {		
+		int topN = 20;
```
Comment (lintool) [inline src/main/java/io/anserini/nrts/TweetServlet.java]: Would be nice to have this parameter be configurable...
Either do it now, or add a // TODO comment to the effect.

**Comment by lintool** [comment]
Looks good, please address issues.

File: src/main/java/io/anserini/nrts/TweetServlet.java (hunk 4 above)
Comment (xeniaqian94) [inline src/main/java/io/anserini/nrts/TweetServlet.java]: User query via a web browser can now be at either http://localhost:8080/search?query=birthday (return top 20 by default) or http://localhost:8080/search?query=birthday&top=15 (user specified).

**Pull request merged**